src/bdk_cli.rs

```diff
@@ -100,34 +109,95 @@ fn main() {
-        tree,
-        AnyBlockchain::from_config(&config).unwrap(),
-    )
-    .unwrap();
+        database,
+        AnyBlockchain::from_config(&config)?,
+    )?;
+    Ok(wallet)
+}
+
+fn new_offline_wallet<D>(
+    network: Network,
+    wallet_opts: &WalletOpts,
+    database: D,
+) -> Result<Wallet<(), D>, Error>
+where
+    D: BatchDatabase,
+{
+    let descriptor = wallet_opts.descriptor.as_str();
+    let change_descriptor = wallet_opts.change_descriptor.as_deref();
+    let wallet = Wallet::new_offline(descriptor, change_descriptor, network, database)?;
+    Ok(wallet)
+}
+
+fn main() {
+    env_logger::init();
 
-    let wallet = Arc::new(wallet);
+    let cli_opts: CliOpts = CliOpts::from_args();
+
+    let network = cli_opts.network;
+    debug!("network: {:?}", network);
+    if network == Network::Bitcoin {
+        warn!("This is experimental software and not currently recommended for use on Bitcoin mainnet, proceed with caution.")
+    }
+
+    let result = match cli_opts.subcommand {
+        CliSubCommand::Wallet {
+            wallet_opts,
+            subcommand: WalletSubCommand::OnlineWalletSubCommand(online_subcommand),
+        } => {
+            let database = open_database(&wallet_opts);
+            let wallet = new_online_wallet(network, &wallet_opts, database).unwrap();
+            let result = bdk_cli::handle_online_wallet_subcommand(&wallet, online_subcommand);
+            serde_json::to_string_pretty(&result.unwrap()).unwrap()
+        }
+        CliSubCommand::Wallet {
+            wallet_opts,
+            subcommand: WalletSubCommand::OfflineWalletSubCommand(offline_subcommand),
+        } => {
+            let database = open_database(&wallet_opts);
+            let wallet = new_offline_wallet(network, &wallet_opts, database).unwrap();
+            let result = bdk_cli::handle_offline_wallet_subcommand(&wallet, offline_subcommand);
+            serde_json::to_string_pretty(&result.unwrap()).unwrap()
+        }
+        CliSubCommand::Key {
+            subcommand: key_subcommand,
+        } => {
+            let result = bdk_cli::handle_key_subcommand(network, key_subcommand);
+            serde_json::to_string_pretty(&result.unwrap()).unwrap()
+        }
+        CliSubCommand::Repl { wallet_opts } => {
+            let database = open_database(&wallet_opts);
+            let online_wallet = new_online_wallet(network, &wallet_opts, database.clone()).unwrap();
+            let online_wallet = Arc::new(online_wallet);
+
+            let offline_wallet = new_offline_wallet(network, &wallet_opts, database).unwrap();
+            let offline_wallet = Arc::new(offline_wallet);
 
-    match cli_opt.subcommand {
-        WalletSubCommand::Repl => {
             let mut rl = Editor::<()>::new();
 
             // if rl.load_history("history.txt").is_err() {
             //     println!("No previous history.");
             // }
 
+            let split_regex = Regex::new(r#"[\w\-]+|"[\w\s]*""#).unwrap();
```

Comment:
If I understand this correctly, the goal here is to split at every space except when the space is contained in the double quotes, right?

Reply:
And what does the `filter_regex` do then?

Reply:
Oh, is it to remove the quotes?

Reply:
Hi yes the filter regex removes the surrounding quotes